phoenix-core/src/main/java/org/apache/phoenix/iterate/MapReduceParallelScanGrouper.java

```diff
@@ -71,6 +72,9 @@ public boolean shouldStartNewScan(QueryPlan plan, List<Scan> scans,
 				FileSystem fs = rootDir.getFileSystem(conf);
 				Path snapshotDir = SnapshotDescriptionUtils.getCompletedSnapshotDir(snapshotName, rootDir);
 				SnapshotDescription snapshotDescription = SnapshotDescriptionUtils.readSnapshotInfo(fs, snapshotDir);
+				//Performing snapshot restore which will be used during scans
```

Comment:
A bit strange for a function called getRegionBoundaries to have side effects, especially when it won't in the non-MR case. I don't see an obvious better place to put this, because the calling function is the same whether we're doing MR or non-MR, I believe. But if we find a good place to do the cleanup (see below), it might also be a better place to do this setup.

Reply:
Does ParallelScanGrouper need a setup and a cleanup hook?